include/myst/options.h

```diff
@@ -9,12 +9,6 @@
 #include <myst/kernel.h>
 #include <myst/types.h>
 
-typedef struct _myst_mount_mapping
-{
-    char** mounts; /* array of source=target */
-    size_t mounts_count;
-} myst_mount_mapping_t;
-
 typedef struct myst_options
```

Comment:
This structure is passed to ``myst_enter_ecall()`` but it contains ``mount_mapping``, which contains strings, which is a security concern if the pointers were actually used.

Reply:
This is exactly what this PR is addressing :)